newrelic/api/transaction.py

```diff
@@ -1813,6 +1813,13 @@ def add_custom_parameters(items):
     return add_custom_attributes(items)
 
 
+def set_user_id(user_id):
+    transaction = current_transaction()
+    if not user_id or not transaction:
+        return
+
+    transaction._add_agent_attribute("enduser.id", user_id)
```

Comment:
Do we need to process this with the attribute system to ensure the attribute gets properly truncated? Can we add a truncation test to check? Maybe add that to the parametrization matrix.